cdap-app-fabric/src/main/java/io/cdap/cdap/internal/provision/DefaultSystemProvisionerContext.java

```diff
@@ -57,8 +61,11 @@ public Map<String, String> getProperties() {
 
   @Override
   public synchronized void reloadProperties() {
-    cConf.reloadConfiguration();
-    properties.set(Collections.unmodifiableMap(cConf.getPropsWithPrefix(prefix)));
+    if (reloadInterval <= 0 || System.currentTimeMillis() > lastReload + reloadInterval) {
```

Comment:
Just a minor issue, but usually I would favor for `-` instead of `+` to avoid overflow (e.g. if `reloadInterval` is set to max long).

`System.currentTimeMillis() - lastReload > reloadInterval`

Reply:
Done